app/models/miq_server/worker_management/systemd.rb

```diff
@@ -3,6 +3,20 @@ def sync_from_system
     self.miq_services = systemd_services.select { |unit| manageiq_service?(unit) }
   end
 
+  def sync_starting_workers!(starting)
+    sync_from_system
+    starting.each do |worker|
+      next if worker.class.rails_worker?
+
+      miq_services_by_guid = miq_services.index_by { |w| w[:name].match(/^.+@(?<guid>.+)\.service$/).named_captures["guid"] }
```

Comment:
I was thinking we could build this in `sync_from_system` after we build the `miq_services` list (on line 3 above) that way we don't have to rebuild it every iteration through the loop here.

Reply:
So add `miq_services_by_guid` as an accessor down on line 36, then in `sync_from_system` you can `self.miq_services_by_guid = miq_services.index_by { |w| w[:name].match(/^.+@(?<guid>.+)\.service$/).named_captures["guid"] }`